homeassistant/components/iron_os/coordinator.py

```diff
@@ -80,3 +122,24 @@ async def _async_update_data(self) -> GitHubReleaseModel:
             assert release.data
 
         return release.data
+
+
+class IronOSSettingsCoordinator(IronOSBaseCoordinator):
+    """IronOS coordinator."""
+
+    def __init__(self, hass: HomeAssistant, device: Pynecil) -> None:
+        """Initialize IronOS coordinator."""
+        super().__init__(hass, device=device, update_interval=SCAN_INTERVAL_SETTINGS)
+
+    async def _async_update_data(self) -> SettingsDataResponse:
+        """Fetch data from Device."""
+
+        characteristics = set(self.async_contexts())
```

Comment:
This will always be empty when booted for the first time no?

Reply:
yes, that's why I request refresh when entities are added. The alternative is fetch all on the device available characteristics but that takes quite long